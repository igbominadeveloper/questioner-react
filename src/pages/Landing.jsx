import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';

//components
import Layout from '../components/presentationals/Layout';
import MeetupCard from '../components/presentationals/MeetupCard/MeetupCard';
import Loader from '../components/presentationals/Loader/Loader';

//stylesheets
import '../assets/css/LandingPage.css';

//modules
import { getUpcomingMeetups } from '../store/modules/meetup';

export class HomePage extends Component {
  componentDidMount() {
    const { upcomingMeetups } = this.props;
    if (Object.keys(upcomingMeetups).length < 1) {
      this.props.getUpcomingMeetups();
    }
  }

  render() {
    let content;
    const { upcomingMeetups, isAuthenticated, isLoading } = this.props;
    {
      isLoading
        ? (content = <Loader />)
        : (content = (
            <Layout>
              <div>
                <header className="banner mt-0" data-test="header" />
                <div className="banner-text flex flex-column">
                  <div className="view-more flex flex-column banner-buttons center mt-120">
                    <p
                      className="text-heavy banner-title"
                      data-test="banner-title"
                    >
                      Questioner
                    </p>
                    <p className="banner-body">
                      Crowd-sourcing questions for meetups
                    </p>
                  </div>
                  <div className="view-more flex flex-row banner-buttons center">
                    {isAuthenticated ? (
                      <Link to="#" data-test="getting-started">
                        <div className="btn-primary shadow">Get Started</div>
                      </Link>
                    ) : (
                      <Link to="/signup" data-test="signup">
                        <div className="btn-primary shadow">Sign up</div>
                      </Link>
                    )}
                  </div>
                </div>
                <div className="view-more">
                  <div className="meetups-heading">Upcoming Meetups</div>
                </div>
              </div>
              <div className="meetup-wrapper">
                {upcomingMeetups &&
                  upcomingMeetups.length > 0 &&
                  upcomingMeetups.map(meetup => (
                    <Link to={`/meetup=${meetup.id}`} key={meetup.id}>
                      <MeetupCard values={meetup} key={meetup.id} />
                    </Link>
                  ))}
              </div>
              <div className="view-more" />
              <div className="how-to flex-column text-white">
                <div className="how-to-title">How Questioner works</div>
                <div className="how-to-details flex flex-row center">
                  <article className="flex width-40 space-around">
                    <div className="icon">
                      <i className="fa fa-2x fa-question text-primary" />
                    </div>
                    <div className="content">
                      <p className="text-left">Find any interesting one?</p>
                      <p className="text-small text-left">
                        Ask your pertinent questions and upvote any to catch the
                        organizer's attention
                      </p>
                      <p className="text-left text-primary text">Sign up</p>
                    </div>
                  </article>
                  <article className="flex width-40 space-around">
                    <div className="icon">
                      <i className="fa fa-2x fa-check text-primary" />
                    </div>
                    <div className="content">
                      <p className="text-left">Ready to book a place?</p>
                      <p className="text-small text-left">
                        You can rsvp on any meetup, better you can change your
                        decision anytime
                      </p>
                      <p className="text-left text-primary text">
                        Book a place
                      </p>
                    </div>
                  </article>
                </div>
              </div>
              <div className="above-footer flex center">
                {isAuthenticated ? (
                  <Link to="#" data-test="getting-started">
                    <div className="view-more flex flex-row banner-buttons space-between">
                      <div className="btn-primary shadow">Get Started</div>
                    </div>
                  </Link>
                ) : (
                  <Link to="/signup" data-test="signup">
                    <div className="view-more flex flex-row banner-buttons space-between">
                      <div className="btn-primary shadow">Sign up</div>
                    </div>
                  </Link>
                )}
              </div>
            </Layout>
          ));
    }
    return content;
  }
}

const mapStateToProps = state => ({
  isLoading: state.meetup.isLoading,
  upcomingMeetups: state.meetup.upcomingMeetups,
  isAuthenticated: state.auth.token !== null,
});

export default connect(
  mapStateToProps,
  { getUpcomingMeetups },
)(HomePage);
