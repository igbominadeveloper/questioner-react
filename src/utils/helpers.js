import jwt from 'jsonwebtoken';

export const decodeToken = token => {
  return jwt.decode(token);
};

export const setItem = (name, value) => {
  localStorage.setItem(name, value);
};

export const getItem = name => {
  const item = localStorage.getItem(name);
  return item;
};

export const destroyItem = item => {
  return localStorage.removeItem(item);
};

export const clearLocalStorage = () => {
  return localStorage.clear();
};
