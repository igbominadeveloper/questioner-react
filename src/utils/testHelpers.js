import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

const mockStore = configureStore([thunk]);

export const setupStore = initialState => mockStore(initialState);

export const findByTestAttribute = (component, attribute) => {
  const wrapper = component.find(`[data-test='${attribute}']`);
  return wrapper;
};
