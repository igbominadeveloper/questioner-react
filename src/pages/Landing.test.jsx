import React from 'react';
import { shallow } from 'enzyme';

import Landing from './Landing';

import { findByTestAttribute } from '../utils/testHelpers';

const component = shallow(<Landing />);

describe('Custom Search', () => {
  it('renders without errors', () => {
    expect(component).toMatchSnapshot();
  });

  it('renders the header section', () => {
    const wrapper = findByTestAttribute(component, 'header');
    expect(wrapper.length).toBe(1);
  });

  it('renders the banner', () => {
    const wrapper = findByTestAttribute(component, 'banner-title');
    expect(wrapper.length).toBe(1);
  });

  it('renders the signup link on the banner', () => {
    const wrapper = findByTestAttribute(component, 'signup');
    expect(wrapper.length).toBe(1);
  });
});
