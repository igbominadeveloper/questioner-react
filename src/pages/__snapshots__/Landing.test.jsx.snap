// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Custom Search renders without errors 1`] = `
<ConnectFunction>
  <div>
    <header
      className="banner mt-0"
      data-test="header"
    />
    <div
      className="banner-text flex flex-column"
    >
      <div
        className="view-more flex flex-column banner-buttons center mt-120"
      >
        <p
          className="text-heavy banner-title"
          data-test="banner-title"
        >
          Questioner
        </p>
        <p
          className="banner-body"
        >
          Crowd-sourcing questions for meetups
        </p>
      </div>
      <div
        className="view-more flex flex-row banner-buttons center"
      >
        <Link
          data-test="signup"
          to="signup.html"
        >
          <div
            className="btn-primary shadow"
          >
            Sign up
          </div>
        </Link>
      </div>
    </div>
    <div
      className="view-more"
    >
      <div
        className="meetups-heading"
      >
        Latest Meetups
      </div>
    </div>
  </div>
  <div
    className="meetup-wrapper"
  >
    <Link
      to="/meetup.html"
    >
      <div
        className="meetup shadow pb-2"
      >
        <div
          className="meetup-image"
          id="laravel"
        />
        <p
          className="meetup-title"
        >
          Laravel Nigeria Meetup
        </p>
        <p
          className="meetup-date text-light"
        >
          <i
            className="fa fa-calendar"
          />
          Thursday, December 20, 9:00 AM
        </p>
        <p
          className="meetup-location text-light"
        >
          <i
            className="fa fa-map-marker"
          />
          Zone Tech Park, Gbagada, Lagos
        </p>
      </div>
    </Link>
    <Link
      to="/meetup.html"
    >
      <div
        className="meetup shadow pb-2"
      >
        <div
          className="meetup-image"
          id="forloop"
        />
        <p
          className="meetup-title"
        >
          ForLoop Ilorin
        </p>
        <p
          className="meetup-date text-light"
        >
          <i
            className="fa fa-calendar"
          />
          Thursday, December 20, 9:00 AM
        </p>
        <p
          className="meetup-location text-light"
        >
          <i
            className="fa fa-map-marker"
          />
          Zone Tech Park, Gbagada, Lagos
        </p>
      </div>
    </Link>
    <Link
      to="/meetup.html"
    >
      <div
        className="meetup shadow pb-2"
      >
        <div
          className="meetup-image"
          id="mocha"
        />
        <p
          className="meetup-title"
        >
          Mocha Akure
        </p>
        <p
          className="meetup-date text-light"
        >
          <i
            className="fa fa-calendar"
          />
          Thursday, December 20, 9:00 AM
        </p>
        <p
          className="meetup-location text-light"
        >
          <i
            className="fa fa-map-marker"
          />
          Zone Tech Park, Gbagada, Lagos
        </p>
      </div>
    </Link>
    <Link
      to="/meetup.html"
    >
      <div
        className="meetup shadow pb-2"
      >
        <div
          className="meetup-image"
          id="vue"
        />
        <p
          className="meetup-title"
        >
          Vue JS Lagos Meetup
        </p>
        <p
          className="meetup-date text-light"
        >
          <i
            className="fa fa-calendar"
          />
          Thursday, December 20, 9:00 AM
        </p>
        <p
          className="meetup-location text-light"
        >
          <i
            className="fa fa-map-marker"
          />
          Zone Tech Park, Gbagada, Lagos
        </p>
      </div>
    </Link>
    <Link
      to="/meetup.html"
    >
      <div
        className="meetup shadow pb-2"
      >
        <div
          className="meetup-image"
          id="laravel"
        />
        <p
          className="meetup-title"
        >
          Laravel Nigeria Meetup
        </p>
        <p
          className="meetup-date text-light"
        >
          <i
            className="fa fa-calendar"
          />
          Thursday, December 20, 9:00 AM
        </p>
        <p
          className="meetup-location text-light"
        >
          <i
            className="fa fa-map-marker"
          />
          Zone Tech Park, Gbagada, Lagos
        </p>
      </div>
    </Link>
    <Link
      to="/meetup.html"
    >
      <div
        className="meetup shadow pb-2"
      >
        <div
          className="meetup-image"
          id="mocha"
        />
        <p
          className="meetup-title"
        >
          Mocha Akure
        </p>
        <p
          className="meetup-date text-light"
        >
          <i
            className="fa fa-calendar"
          />
          Thursday, December 20, 9:00 AM
        </p>
        <p
          className="meetup-location text-light"
        >
          <i
            className="fa fa-map-marker"
          />
          Zone Tech Park, Gbagada, Lagos
        </p>
      </div>
    </Link>
    <Link
      to="/meetup.html"
    >
      <div
        className="meetup shadow pb-2"
      >
        <div
          className="meetup-image"
          id="vue"
        />
        <p
          className="meetup-title"
        >
          Vue JS Lagos Meetup
        </p>
        <p
          className="meetup-date text-light"
        >
          <i
            className="fa fa-calendar"
          />
          Thursday, December 20, 9:00 AM
        </p>
        <p
          className="meetup-location text-light"
        >
          <i
            className="fa fa-map-marker"
          />
          Zone Tech Park, Gbagada, Lagos
        </p>
      </div>
    </Link>
  </div>
  <div
    className="view-more"
  />
  <div
    className="how-to flex-column text-white"
  >
    <div
      className="how-to-title"
    >
      How Questioner works
    </div>
    <div
      className="how-to-details flex flex-row center"
    >
      <article
        className="flex width-40 space-around"
      >
        <div
          className="icon"
        >
          <i
            className="fa fa-2x fa-question text-primary"
          />
        </div>
        <div
          className="content"
        >
          <p
            className="text-left"
          >
            Find any interesting one?
          </p>
          <p
            className="text-small text-left"
          >
            Ask your pertinent questions and upvote any to catch the organizer's attention
          </p>
          <p
            className="text-left text-primary text"
          >
            Sign up
          </p>
        </div>
      </article>
      <article
        className="flex width-40 space-around"
      >
        <div
          className="icon"
        >
          <i
            className="fa fa-2x fa-check text-primary"
          />
        </div>
        <div
          className="content"
        >
          <p
            className="text-left"
          >
            Ready to book a place?
          </p>
          <p
            className="text-small text-left"
          >
            You can rsvp on any meetup, better you can change your decision anytime
          </p>
          <p
            className="text-left text-primary text"
          >
            Book a place
          </p>
        </div>
      </article>
    </div>
  </div>
  <div
    className="above-footer flex"
  >
    <div
      className="view-more flex flex-row banner-buttons space-between"
    >
      <div
        className="btn-primary shadow"
      >
        Sign up
      </div>
    </div>
  </div>
</ConnectFunction>
`;
